File: 169_majority_algorithm.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        idx=0;
        count=1;
        
        for num in range(1,len(nums)):
            if(nums[num]==nums[idx]):
                count+=1
            else:
                count-=1
            if(count==0):
                idx=num
                count=1
                
            
        return nums[idx]
                
        

File: 169_majority_algorithm.java
class Solution {
    public int majorityElement(int[] nums) {
         
        int idx=0;
        int count=1;
        
        for(int i=1; i<nums.length;i++){
            if(nums[i]==nums[idx]){
                count++;
            }else{
                count--;
            }
            if(count==0){
                idx=i;
                count=1;
            }
        }
       
      return nums[idx];
    }
}
